### PR TITLE
adds preliminary support for Neo4j

### PR DESCRIPTION
**Pull request by tuddman**
neo4j+bolt works with `:jdbc-url` explicitly set. 
works for my use-case

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [![hikari-cp](http://clojars.org/hikari-cp/latest-version.svg)](http://clojars.org/hikari-cp)
 
 # hikari-cp [![Circle CI](https://circleci.com/gh/tomekw/hikari-cp/tree/master.png?style=badge&circle-token=a17dfed149321f3a2e9b9af11cea96b9003a047a)](https://circleci.com/gh/tomekw/hikari-cp/tree/master)
-
+j
```
Comment (tomekw) [inline README.md]: @tuddman hm? :)

File: README.md [hunk 2]
```diff
@@ -78,6 +78,7 @@ Custom translations of properties can be added by extending the
 | `hsqldb`         | `org.hsqldb.jdbc.JDBCDataSource`                   | No                    |
 | `mariadb`        | `org.mariadb.jdbc.MySQLDataSource`                 | No                    |
 | `mysql`          | `com.mysql.jdbc.jdbc2.optional.MysqlDataSource`    | **Yes**               |
+| `neo4j`          | `org.neo4j.jdbc.DataSource`										    | No			              |
```
Comment (tomekw) [inline README.md]: Weird formatting?

File: README.md [hunk 3]
```diff
@@ -121,6 +122,36 @@ Custom translations of properties can be added by extending the
...
+
+(def datasource-options {:adapter            "neo4j:bolt" ; neo4j:bolt+routing
+                         :username           "username"
+                         :password           "password"
+                         :port-number        7687
+                         :withMaxConnectionLifetime 1800000 
+                         :withMaxConnectionPooolSize 50
+                         :withConnectionAcquisitionTimeout 120000 
+                         })
+
```
Comment (tomekw) [inline README.md]: Weird formatting, too many empty lines.

File: README.md [hunk 4]
```diff
@@ -205,3 +236,4 @@ Copyright © 2014 - 2017 [Tomek Wałkuski](https://github.com/tomekw) and [Jan S
 
 Distributed under the Eclipse Public License either version 1.0 or (at
 your option) any later version.
+                         :port-number        7687
```
Comment (tomekw) [inline README.md]: Line not needed?

File: project.clj [hunk 5]
```diff
@@ -1,4 +1,4 @@
-(defproject hikari-cp "2.0.1"
+(defproject hikari-cp "2.0.2-SNAPSHOT"
```
Comment (tomekw) [inline project.clj]: Please leave project version as it was :)

File: README.md [hunk 6]
```diff
@@ -121,6 +122,36 @@ Custom translations of properties can be added by extending the
...
+                         :withMaxConnectionLifetime 1800000 
+                         :withMaxConnectionPooolSize 50
+                         :withConnectionAcquisitionTimeout 120000 
+                         })
+
+
+;; OR explicitly
+(def datasource-options {:jdbc-url "jdbc:neo4j:bolt://host:port/?username=neo4j,password=xxxx,debug=true" })
+
+(def datasource
```
Comment (tomekw) [inline README.md]: Fits one line?

File: README.md [hunk 7]
```diff
@@ -121,6 +122,36 @@ Custom translations of properties can be added by extending the
...
+  (make-datasource datasource-options))
+
+(defn -main [& args] 
+  (jdbc/with-db-connection [conn {:datasource datasource}]
+    (let [rows (jdbc/query conn "MATCH (n:Foo) RETURN n")]
+      (println rows)))
+  (close-datasource datasource))
+```
+
```
Comment (tomekw) [inline README.md]: Too many empty lines.

File: project.clj (hunk 5 above)
Comment (tuddman) [reply project.clj]: was following convention of bumping patch version for any new changes beyond the released *.jar

I have reverted `project.clj` back to `2.0.1`

File: README.md (hunk 1 above)
Comment (tuddman) [reply README.md]: removed.

File: README.md (hunk 2 above)
Comment (tuddman) [reply README.md]: fixed.

File: README.md (hunk 3 above)
Comment (tuddman) [reply README.md]: removed.

File: README.md (hunk 6 above)
Comment (tuddman) [reply README.md]: `jdbc-url` connection string fits in one line, with all parameters passed in.

File: README.md (hunk 7 above)
Comment (tuddman) [reply README.md]: removed

File: README.md (hunk 4 above)
Comment (tuddman) [reply README.md]: removed

**Comment by tuddman** [comment]
I don't think `https://github.com/neo4j-contrib/neo4j-jdbc/blob/3.1/neo4j-jdbc/src/main/java/org/neo4j/jdbc/DataSource.java` is correct for setting the DataSource to `org.neo4j.jdbc.DataSource` 

so not working with using `:adapter`

does work with `:jdbc-url`

**Comment by jstepien** [comment]
LGTM. Thanks for the patch!

**Pull request merged**